src/main/java/org/minions/devfund/damianvp/WhackAMole.java

```diff
@@ -0,0 +1,178 @@
+     * @param gridDimension dimension of grid game
+     */
+    WhackAMole(int numAttempts, int gridDimension) {
+        this.attemptsLeft = numAttempts;
+        this.moleGrid = initializeGrid(gridDimension);
+        this.trackerMoleGrid = initializeGrid(gridDimension);
+        this.molesLeft = 0;
+        this.score = 0;
+    }
+
+    /**
+     * Initialize the grid game with empty values.
+     * @param gridDimension  dimension of grid to build.
+     * @return a grid loaded with empty values.
+     */
+    private char[][] initializeGrid(int gridDimension) {
+        char[][] grid = new char[gridDimension][gridDimension];
+        for (char[] row: grid) {
+            Arrays.fill(row, EMPTY);
+        }
+        return grid;
+    }
+
+    /**
+     * Method to place a specific mole by Location x, y.
+     * @param x  x location of grid.
+     * @param y  y location of grid.
+     * @return  true, if place is successful.
+     *          false, if place was not performed.
+     */
+    boolean place(int x, int y) {
+        if (validatedCoordinates(x, y) && this.moleGrid[x][y] != MOLE) {
+            this.moleGrid[x][y] = MOLE;
+            this.molesLeft++;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Method to randomly place n quantity of moles.
+     * @param moles quantity of moles to place randomly.
+     */
+    void randomPlace(int moles) {
+        Random random = new Random();
+        int max = this.moleGrid.length;
+        int min = 0;
+        boolean moleIsPlaced;
+        if (moles <= this.moleGrid.length * this.moleGrid.length) {
+            while (moles > 0) {
+                int x = random.nextInt(max - min);
+                int y = random.nextInt(max - min);
+                moleIsPlaced = place(x, y);
+                if (moleIsPlaced) {
+                    moles--;
+                }
+            }
+        }
+        System.out.println("Is not possible place more than fdsfsdf");
+    }
+
+    /**
+     * Method to whack a location in order to search a mole.
+     * @param x x location of grid game.
+     * @param y y location of grid game
+     */
+    void whack(int x, int y) {
+        if (validatedCoordinates(x, y)) {
+            if (this.moleGrid[x][y] == MOLE) {
+                System.out.println("Good, you found a mole!!");
+                this.molesLeft--;
+                this.score++;
+
+                this.trackerMoleGrid[x][y] = WHACKED;
+                this.moleGrid[x][y] = WHACKED;
+            }
+            this.attemptsLeft--;
+            verifyGameStatus();
+        }
+    }
+
+    /**
+     * Method to print the grid available for user.
+     */
+    void printGridToUser() {
+        System.out.println(Arrays.deepToString(this.trackerMoleGrid));
+    }
+
+    /**
+     * Method to print the complete grid game status.
+     */
+    void printGrid() {
+        System.out.println(Arrays.deepToString(this.moleGrid));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)